spec/models/access_spec.rb

```diff
@@ -0,0 +1,636 @@
+require "rails_helper"
+
+RSpec.describe Access, type: :model do
+  describe "Associations" do
+    it { is_expected.to belong_to(:user) }
+    pending { is_expected.to belong_to(:resource).optional }
```

Comment:
Need to figure out this test, it's a little tricky because our resources are currently _optionally optional_ 🧠

Reply:
You can set up separate subjects for separate tests.

```ruby
context "for super admins" do
  subject { Access.new(role: :super_admin) } 
  it { is_expected.not_to belong_to(:resource) } # I think you can do negative matchers?
end

context "for everyone else" do
  subject { Access.new(role: :viewer) } 
  it { is_expected.to belong_to(:resource) } # I think you can do negative matchers?
end
```

Would that work?

Reply:
Ah nice, let me try this, I had no idea it was picking up `subject` like that.

Reply:
Hehe yeah it's pretty neat. If unspecified, it spins up a `model.new` as the subject by default.

Reply:
I got this working, but in a slightly different, more explicit way. The problem with that spec helper is that it doesn't understand our nuanced constraint and throws up this error:

```
Expected Access to have a belongs_to association called resource (and for the record not to fail validation if :resource is unset; i.e., either the association should have been defined with `optional: true`, or there should not be a presence validation on :resource)
```

Basically, it doesn't allow you to use `belongs_to` having both an `optional` and a `presence` validation together. Which we do, but conditionally so. Hence, it's not technically incorrect, just that the spec helper isn't smart enough to determine that conditional.

So I re-wrote it in a more straightforward way [here](https://github.com/simpledotorg/simple-server/pull/1185/commits/e9aa57aa00d4fe38c776d98256b3e694cea33687).